### DISCUSSION TITLE
Unable to load the website

### DISCUSSION THREAD

**Discussion by REGEX777** [pretix/pretix#2747] [Q&A]
Hey, so i have 0 experience in nginx configuration.
 
when I use this config

`server {
    listen 80 default_server;
    listen [::]:80 ipv6only=on default_server;
    server_name events.aucrogroup.com;
}
server {
    listen 443 default_server;
    listen [::]:443 ipv6only=on default_server;
    server_name events.aucrogroup.com;

    ssl on;
    ssl_certificate /etc/letsencrypt/live/events.aucrogroup.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/events.aucrogroup.com/privkey.pem;

    add_header Referrer-Policy same-origin;
    add_header X-Content-Type-Options nosniff;

    location / {
        proxy_pass http://localhost:8345;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Host $http_host;
    }

    location /media/ {
        alias /var/pretix/data/media/;
        expires 7d;
        access_log off;
    }

    location ^~ /media/cachedfiles {
        deny all;
        return 404;
    }
    location ^~ /media/invoices {
        deny all;
        return 404;
    }

    location /static/ {
        alias /var/pretix/venv/lib/python3.10/site-packages/pretix/static.dist/;
        access_log off;
        expires 365d;
        add_header Cache-Control "public";
    }
}`

I get the default nginx page
please help me figure out what am I doing wrong

**Comment by raphaelm** [comment]
The upper config block, for http/port 80 does not do anything and is probably missing a redirect, the lower config block for https/port 443 is looking quite good.

If https also doesn't work, the file might be in the wrong directory or conflicting with another config file.

**Comment by REGEX777** [comment]
Hey @raphaelm,
Thanks for your reply appreciate it.

are you talking about
`location /static/ {
    alias /var/pretix/venv/lib/python3.10/site-packages/pretix/static.dist/;
    access_log off;
    expires 365d;
    add_header Cache-Control "public";
}
`
 this?

**Reply by raphaelm** [reply]
No, the server{} block at the top. Seems there's a problem with our docs maybe. Are you accessing the page via http:// or https://? Are you getting the nginx page in both cases?

**Reply by raphaelm** [reply]
If fixed a documentation issue, apparently the redirect from http to https was missing in the recommended configuration. That should not be the problem if you access it via https, though
https://github.com/pretix/pretix/commit/9ce519bb0b37b53fb0b436964aee655603fde414

**Reply by REGEX777** [reply]
Hey @raphaelm, 
Thanks again! Yes im getting the nginx default page on both

**Reply by REGEX777** [reply]
I will try it and will share the update with you thanks 😄

**Reply by REGEX777** [reply]
hey @raphaelm i tried to update the config but now nothing is listening to port 80 and 443 I tried change ipv6only to ipv4only but that didn't worked here's my updated config

```
`
server {
    listen 80 default_server;
    listen [::]:80 ipv4only=on default_server;
    server_name events.aucrogroup.com;
    location / {
        return 301 https://$host$request_uri;
    }
}
server {
    listen 443 default_server;
    listen [::]:443 ipv4only=on default_server;
    server_name events.aucrogroup.com;

    ssl on;
    ssl_certificate /etc/letsencrypt/live/events.aucrogroup.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/events.aucrogroup.com/privkey.pem;

    add_header Referrer-Policy same-origin;
    add_header X-Content-Type-Options nosniff;

    location / {
        proxy_pass http://localhost:8345;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto https;
        proxy_set_header Host $http_host;
    }

    location /media/ {
        alias /var/pretix/data/media/;
        expires 7d;
        access_log off;
    }

    location ^~ /media/cachedfiles {
        deny all;
        return 404;
    }
    location ^~ /media/invoices {
        deny all;
        return 404;
    }

    location /static/ {
        alias /var/pretix/venv/lib/python3.8/site-packages/pretix/static.dist/;
        access_log off;
        expires 365d;
        add_header Cache-Control "public";
    }
}

`
```

**Reply by raphaelm** [reply]
Can you post the output of

systemctl status nginx

and

netstat -nplt

to check if nginx is running and where it is listening?

**Reply by REGEX777** [reply]
output of the nginx status command

```
(venv) pretix@vps-4396535b:/etc/nginx/sites-available$ systemctl status nginx
● nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
   Active: active (running) since Tue 2022-08-02 16:07:48 UTC; 2min 58s ago
     Docs: man:nginx(8)
  Process: 14222 ExecStartPre=/usr/sbin/nginx -t -q -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
  Process: 14223 ExecStart=/usr/sbin/nginx -g daemon on; master_process on; (code=exited, status=0/SUCCESS)
 Main PID: 14224 (nginx)
    Tasks: 3 (limit: 4603)
   Memory: 3.5M
   CGroup: /system.slice/nginx.service
           ├─14224 nginx: master process /usr/sbin/nginx -g daemon on; master_process on;
           ├─14225 nginx: worker process
           └─14226 nginx: worker process
(venv) pretix@vps-4396535b:/etc/nginx/sites-available$ 
```

output of netstat -nplt command

```
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:5432          0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:8345          0.0.0.0:*               LISTEN      12289/python3       
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN      -                   
tcp        0      0 127.0.0.1:6379          0.0.0.0:*               LISTEN      -                   
tcp6       0      0 :::22                   :::*                    LISTEN      -                   
tcp6       0      0 ::1:5432                :::*                    LISTEN      -                   
tcp6       0      0 ::1:25                  :::*                    LISTEN      -                   
tcp6       0      0 ::1:6379                :::*                    LISTEN      -                   
```

**Reply by raphaelm** [reply]
This is very weird, something is very wrong with your nginx setup (certainly unrelated to pretix) and I don't know what. Maybe some config files in the wrong folder or something like that. I don't think I can help debug this well.

**Reply by REGEX777** [reply]
umm T_T

**Reply by Dylaneux** [reply]
Any other developers on here maybe know what this issue could be? Nginx is working perfectly fine and displays the 'Welcome to nginx!' screen but when the Pretix installation has been completed, nothing displays at all.

**Reply by raphaelm** [reply]
> Any other developers on here maybe know what this issue could be? Nginx is working perfectly fine and displays the 'Welcome to nginx!' screen but when the Pretix installation has been completed, nothing displays at all.

That does not seem to be the case any longer, nginx does not seem to work at all if netstat command does not list it

**Reply by Dylaneux** [reply]
What do you think the issue is?

**Reply by Dylaneux** [reply]
Do you think there is a resolution to this? We previously used Ubuntu 22.04 and we tried to reinsrtall Pretix on Debian with docker but same issues :( So we've reinstalled the operating system a good few times.

**Reply by raphaelm** [reply]
Unfortunately so far I could not guess what you are doing wrong.

**Reply by Dylaneux** [reply]
Ahh so you think it could be user error rathe than server error?

**Reply by raphaelm** [reply]
I assume that you made some error in the server configuration, yes, but I can't know for sure.
